### DISCUSSION TITLE
Problem with action/chekout another private repo, PAT not working

### DISCUSSION THREAD

**Discussion by mfn** [community/community#26195] [Actions]
<p>Hi,</p>
<p>I’ve a job in repo_a which also needs a checkout of repo_b.</p>
<p>Both are private repositories within the same organization (call it “the_org”).</p>
<p>I followed allt the guides: <a href="https://github.com/actions/checkout#Checkout-multiple-repos-private" rel="nofollow noopener">https://github.com/actions/checkout#Checkout-multiple-repos-private</a> and there’s also <a href="https://github.community/t/best-way-to-clone-a-private-repo-during-script-run-of-private-github-action/16116">https://github.community/t5/GitHub-Actions/Best-way-to-clone-a-private-repo-during-script-run-of-private/m-p/20221</a> but none of this works.</p>
<p>This is the job definition:</p>
<pre><code class="lang-auto">
- name: Checkout code  
uses: actions/checkout@v2

- name: Checkout repo\_b  
uses: actions/checkout@v2  
with:  
repository: the\_org/repo\_b  
token: ${{ secrets.PAT\_CHECKOUT\_REPO\_B }}  
path: repo\_b

</code></pre>
<p>I made double sure that the secret is there, that it’s named correctly, etc.</p>
<p>I always end up with this error</p>
<pre><code class="lang-auto">
Run actions/checkout@v2  
with:  
repository: the\_org/repo\_b  
path: repo\_b  
persist-credentials: true  
clean: true  
fetch-depth: 1  
lfs: false  
env:  
cache-key: cache-v1  
extensions: bcmath, curl, fpm, gd, intl, mbstring, pgsql, xml, zip  
ini-values: memory\_limit=-1  
Added matchers: 'checkout-git'. Problem matchers scan action output for known warning or error strings and report these inline.  
Syncing repository: the\_org/repo\_b  
Working directory is '/home/runner/work/repo\_a/repo\_a/repo\_b'  
/usr/bin/git version  
git version 2.25.0  
/usr/bin/git init /home/runner/work/repo\_a/repo\_a/repo\_b  
Initialized empty Git repository in /home/runner/work/repo\_a/repo\_a/repo\_b/.git/  
/usr/bin/git remote add origin [https://github.com/the\_org/repo\_b](https://github.com/the_org/repo_b)  
/usr/bin/git config --local gc.auto 0  
/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader  
/usr/bin/git config --local http.[https://github.com/.extraheader](https://github.com/.extraheader) AUTHORIZATION: basic \*\*\*  
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/master:refs/remotes/origin/master  
##[error]fatal: could not read Username for '[https://github.com](https://github.com)': terminal prompts disabled  
The process '/usr/bin/git' failed with exit code 128  
Waiting 17 seconds before trying again  
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/master:refs/remotes/origin/master  
##[error]fatal: could not read Username for '[https://github.com](https://github.com)': terminal prompts disabled  
The process '/usr/bin/git' failed with exit code 128  
Waiting 13 seconds before trying again  
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/master:refs/remotes/origin/master  
##[error]fatal: could not read Username for '[https://github.com](https://github.com)': terminal prompts disabled  
Removed matchers: 'checkout-git'  
##[error]The process '/usr/bin/git' failed with exit code 128

</code></pre>
<p>I tried the PAT locally and it worked, though indeed locally I was asked interactively for a username; I entered mine and the PAT was accepted a the password.</p>
<p>Is there something I’m missing?</p>
<p>thanks,</p>
<ul>
<li>Markus</li>
</ul>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mfn">@mfn</a> ,</p>
<p>Your sample code works fine on my side, it can checkout repo_b without any problem.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181096022-0eb60269-aedc-47db-8a8f-c6d81a0d8553.png" alt="Checkoutrepo_b.png" data-base62-sha1="5eN1b31uw1Pil7guOJxRu1zi56p" width="492" height="221"></p>
<p>I can reproduce your error if i delete the secrets in repo_a. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/23a3ee0804d45320f191cd6173b80954efc72cbb.png" data-download-href="/uploads/short-url/55hVCRbxghO6RWZGkwUUo6SdF7J.png?dl=1" title="Checkouterror.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181096023-0034b0f5-6b9d-4b2b-9ccd-0fb6adbe9611.png" alt="Checkouterror.png" data-base62-sha1="55hVCRbxghO6RWZGkwUUo6SdF7J" width="690" height="383" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/23a3ee0804d45320f191cd6173b80954efc72cbb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Checkouterror.png</span><span class="informations">922×512 40.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Hence, please check the secrets setting for repo_a, confirm secrets with correct PAT is created, and there’s not any typo, or you can create a new PAT and set it as secrets for testing.</p>
<p>Thanks.</p>

**Comment by mfn** [comment]
<p>You’re absolutely right, it works as documented / expected.</p>
<p>I’m officially an idiot because I was pre-testing stuff on separate repos first and simply added the secret to the wrong repository :boom:</p>
<p>Curious question: I did try out the PAT on git cli locally to confirm it works. I was asked for a username there and the PAT instead of my password, it worked.</p>
<p>A github checkout action doesn’t have my “username”, why does it work there? Or: why is the username for a PAT not required? Or why is required for the https endpoint? :slight_smile:</p>
<p>Thanks!</p>

**Comment by weide-zhou** [comment]
<p>The config option ‘actions/checkout’ use is the .extraheader property which directly adds the auth header to the http requests in git and that is very likely what is being using to authenticate even if the action the user is trying to use specifies a PAT in the URL. That’s why user name is not needed. Thanks.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/de337db962226b8a2b84373f121d4edcd4b98129.png" data-download-href="/uploads/short-url/vHGjGZ54TtMR8Jrx1GLg7nfI1Qt.png?dl=1" title="extraheader.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181096024-10dcb385-808e-4da6-930c-15b90d4178ae.png" alt="extraheader.png" data-base62-sha1="vHGjGZ54TtMR8Jrx1GLg7nfI1Qt" width="690" height="259" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/de337db962226b8a2b84373f121d4edcd4b98129_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">extraheader.png</span><span class="informations">722×272 19.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
